src/goto-programs/remove_const_function_pointers.h

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************\
+
+Module: Goto Programs
+
+Author: Thomas Kiley, thomas.kiley@diffblue.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_GOTO_PROGRAMS_REMOVE_CONST_FUNCTION_POINTERS_H
+#define CPROVER_GOTO_PROGRAMS_REMOVE_CONST_FUNCTION_POINTERS_H
+
+#include "goto_model.h"
+#include <util/message.h>
+#include <util/expr.h>
+#include <util/namespace.h>
+
+
+class remove_const_function_pointerst:public messaget
+{
+public:
+  typedef std::set<exprt> functionst;
```

Comment:
Well I had thought it was required to make the tests actually test anything, but actually checking for the existence of the goto in the no-match and approx cases will test to the same precision so I will do that.